rules/aip0132/resource_reference_type_test.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package aip0132
+
+import (
+	"testing"
+
+	"github.com/googleapis/api-linter/rules/internal/testutils"
+)
+
+func TestResourceReferenceType(t *testing.T) {
+	// Set up testing permutations.
+	tests := []struct {
+		testName string
+		TypeName string
+		RefType  string
+		problems testutils.Problems
```

Comment:
With the `Message` fields on the `testutils.Problem`, the value used to compare with `Diff(...)` is, `"child_type"` which appears in both `lint.Problem` `Message`s. So the test, can have false results, I think. Please fix if you agree.

Reply:
Yep you are right!